validator/client/beacon-api/submit_aggregate_selection_proof.go

```diff
@@ -72,6 +72,64 @@ func (c *beaconApiValidatorClient) submitAggregateSelectionProof(
 	}, nil
 }
 
+func (c *beaconApiValidatorClient) submitAggregateSelectionProofElectra(
```

Comment:
I'm not sure if there's any clean way to deal with this code duplication

Reply:
You can remove some of the duplication by having some generic func like:
```
func (c *beaconApiValidatorClient) submitAggregateSelectionProofGeneric(ctx context.Context, in *ethpb.AggregateSelectionRequest, committeeLength uint64) ([32]byte, error) {
	isOptimistic, err := c.isOptimistic(ctx)
	if err != nil {
		return [32]byte{}, err
	}

	// An optimistic validator MUST NOT participate in attestation. (i.e., sign across the DOMAIN_BEACON_ATTESTER, DOMAIN_SELECTION_PROOF or DOMAIN_AGGREGATE_AND_PROOF domains).
	if isOptimistic {
		return [32]byte{}, errors.New("the node is currently optimistic and cannot serve validators")
	}

	isAggregator, err := helpers.IsAggregator(committeeLength, in.SlotSignature)
	if err != nil {
		return [32]byte{}, errors.Wrap(err, "failed to get aggregator status")
	}
	if !isAggregator {
		return [32]byte{}, errors.New("validator is not an aggregator")
	}

	attestationData, err := c.attestationData(ctx, in.Slot, in.CommitteeIndex)
	if err != nil {
		return [32]byte{}, errors.Wrapf(err, "failed to get attestation data for slot=%d and committee_index=%d", in.Slot, in.CommitteeIndex)
	}

	attestationDataRoot, err := attestationData.HashTreeRoot()
	if err != nil {
		return [32]byte{}, errors.Wrap(err, "failed to calculate attestation data root")
	}

	return attestationDataRoot, nil
}
```